crypto/pkg/tls/tls.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tls
+
+import (
+	"crypto/tls"
+)
+
+// NewTLSConfig returns a tls.Config with the recommended default settings for Calico Enterprise components.
+// Read more recommendations here:
+// https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4095
+func NewTLSConfig(fipsMode bool) *tls.Config {
+	cfg := &tls.Config{
+		MinVersion: tls.VersionTLS12,
```

Comment:
Is there any reason to believe that we should still be supporting old (and known to be vulnerable) TLS versions?

Reply:
I don't think so since this is just used by our components. It's possible you might need it for an upgrade, but I'm not exactly sure how the upgrade logic works.